### DISCUSSION TITLE
Newlines in base64 attachment

### DISCUSSION THREAD

**Discussion by bulldozer2003** [nextgenhealthcare/connect#5300] [Q&A] [answered]
I am trying to send an attachment as part of a JSON object using the HTTP sender destination. The server kept responding with "invalid JSON" which I determined is due to there being newlines in the base64 string. My HTTP sender is configured to send the test content below where `${attachment}` is a variable containing the `${ATTACH:...}` token string. When I use this, everything is sent as I expect but the base64 contains those additional newlines which breaks the JSON.
```
{
  "foo":"bar",
  "baz":"${attachment}"
}
```

I am loading the attachment from an SMB share using a javascript attachment script. I'm loading the file as a byte array and using `addAttachment(file, 'audio/mpeg', true)`. I don't believe reading the file from SMB would add newlines to an mp3 file arbitrarily.

In trying to troubleshoot where the newlines are coming from I added a javascript writer destination with the following:
```
var attachment = getAttachment($('attachmentId'), false);
logger.info(attachment.getContent());
logger.info(attachment.getContentString());
```

The output of `getContentString()` contains the newlines implying mirth is adding them when I load the file?

We are using mirth version 3.12

**Comment by pacmano1** [comment, marked as answer]
you need to just .replaceAll \n\r on the base64 string.  Yes mirth is adding it.

```JS
getAttachments().get(0).getContentString().replaceAll( "\r\n", "" );
```
or similar of course.

**Reply by bulldozer2003** [reply]
Thank you for the response. I only used `getContentString()` to troubleshoot and forgot to mention my HTTP Sender setup. I'm trying to use the attachment as base64 directly in the HTTP sender content. Below is my testing JSON payload. `${attachment}` is a variable containing the `${ATTACH:...}` token string. When I use this, everything is sent as I expect but the base64 contains those additional newlines which breaks the JSON.
```
{
  "foo":"bar",
  "baz":"${attachment}"
}
```
